pwa-devdocs/_drafts/pwa-studio-fundamentals/extensibility-with-targets/index.md

```diff
@@ -0,0 +1,512 @@
+The `perfReport` target runs its interceptors asynchronously and in parallel.
+This is appropriate for logging and monitoring interceptors that don't affect functionality.
+The `socialIcons` target runs its interceptors synchronously and in subscription order, passing return values as arguments to the next interceptor.
+This is appropriate for customizations that must happen in a predictable order.
+
+Targets are variants of Tapables; see [Tapable](https://github.com/webpack/tapable) for a list of available types and behaviors.
+(Note that the Tapable classnames all end with 'Hook', and the Target classnames do not.)
+
+The BuildBus runs all declare files before running any intercept files, so simply registering a declare file will guarantee that the targets are available to any dependent interceptor.
+Like intercept files, declare files are CommonJS modules that run in Node. Organize them together with intercept files.
+
+To register `./targets/declare.js` as a declare file, add its path to `package.json` under the `pwa-studio.targets.declare` section:
+   ```diff
+      "module": "src/index.js",
+      "es2015": "src/index.js",
+      "esnext": "src/index.js",
+      "pwa-studio": {
+        "targets": {
+   +      "declare": "./targets/declare.js",
+          "intercept": "./targets/intercept.js"
+        }
+      }
+    }
+   ```
+
+# API
+
+## List of targets
+
+<!-- TODO: Adapt or generate this from the JSDoc comments all over all the declare files -->
+
+# Development
+
+## Extension development
+
+When writing a PWA-compatible library, to publish on NPM or on the Magento Marketplace, think of Targets as the main tool in your toolbox.
+To add new functionality, try adding interceptors to your storefront project and injecting the functionality that way.
+When your prototype is working, try separating that work into a separate module and continuing to work on it as an encapsulated, portable extension!
+
+### Initial phase: in-project interceptors
+
+As in the [Quick Start][], new functionality may begin by intercepting from the storefront project itself.
+The tutorial creates a Greeting Page by implementing the page in the storefront project's source folder.
+
+The next step is to turn that functionality into a new module.
+
+### Move the code
+
+Create a new project folder alongside your storefront. Run `yarn init` and then add the `pwa-studio` section to your `package.json`.
+
+```json
+{
+  "name": "@me/pwa-greeting-page",
+  "version": "1.0.0",
+  "description": "Hello, anything!",
+  "author": "PWA Developer",
+  "main": "./GreetingPage.js",
+  "license": "MIT",
+  "pwa-studio": {
+    "targets": {
+      "intercept": "./targets/intercept.js"
+    }
+  }
+}
+```
+
+Create `./targets/intercept.js`. Now that you're working in a third-party module, you can use default NodeJs module resolution in the route path, and make your greeting component the default export of the package.
+Create `packages/venia-concept/targets/local-intercept.js`:
```

Comment:
I believe `targets/intercept` is replacing `local-intercept`

Reply:
I should make them consistent in the document, but @fooman you should know that there's actually no required filename or file structure. You manually reference the path to your intercept file in `package.json`, so it can be anywhere you want. In the future we can add a "preset" to reduce this kind of boilerplate, but for now it's flexible in order to support many project structures and tooling scenarios.

Reply:
This paragraph mentions to create `./targets/intercept.js` and `packages/venia-concept/targets/local-intercept.js` of which only `./targets/intercept.js` would be referenced in the earlier mentioned `packages.json` file.

And yes agree that establishing conventions around patterns from the start would be good. Ie as a developer being able to browse under the targets folder of any package to quickly see what it does would be great.